Terasology/Apiculture
# Add apiary functionality

Migrates bee mating commands to apiary, and implements the life cycle of bees.

Strongly recommended to test alongside SoundyGenetics#3, otherwise drones will stack with potentially genetics erasing effects. The combination of these PRs notably allows drones to stack only if their genetics are the same. Princesses and queens never stack.

The apiary can be interacted with, displaying a UI with two slots arranged vertically on the left, a lifespan bar adjacent to these slots, and a honeycomb of product slots. If a princess is placed in the top left slot and a drone in the bottom left slot, after a time (currently 1 second), indicated by the filling of the lifespan bar in red, they will be mated, with the resulting queen occupying the top left slot.

If a queen occupies the top left slot, it will be active, and after a time being active will have its lifespan tick down, indicated by the emptying of a yellow lifespan bar. Once its life has ticked down completely, it will die, and its offspring, 1 princess and a genetics dependent number of drones, will appear in the output. Additionally, each lifespan tick has a chance of producing a honeycomb in the output.

Testing:
- Ensure that the apiary meets the general description above with no unexpected behavior.
- The top left slot should only accept princesses or queens (not drones or other items).
- The top right slot should only accept drones (not princesses/queens/other items).
- The output slots should only allow the removal of items.
- Closing the UI should not prevent mating or lifespan ticks.
- Changing the item in the princess or drone slot should interrupt mating.
- Placing drones in the bottom left while a queen is in the top left is permitted, but should otherwise have no effect
- Removing a queen and replacing it should not reset its lifespan. (It will however reset the current lifespan tick. This is intended.)
- If there is not a queen present or mating is not occurring, the lifespan bar should be empty.

Advanced Testing:
- Ensure random genetic combination still occurs as expected
- Ensure drone stacking occurs based on genetics
    - Drones spawned using give or from pure bred lines should stack
    - Drones from first-gen cross breeds should stack assuming mutation does not occur
    - Drones from second-gen cross breeds (cross of two first-gen bees) should rarely stack (only 1/16 chance)
    - Obviously, drones of different species should not stack
    - Princesses and queens should not stack
- Ensure that the following traits hold for the three species:
    - Species A - Lifespan: 6 ticks, Tick length: 75 seconds, Production chance: 25% per tick, Drone offspring: 1
    - Species B - Lifespan: 3 ticks, Tick length: 100 seconds, Production chance: 25% per tick, Drone offspring: 2
    - Species C - Lifespan: 9 ticks, Tick length: 50 seconds, Production chance: 50% per tick, Drone offspring: 4
